sdk/storage/blobs/blob_lease_client.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+namespace Azure { namespace Storage {
+
+class BlobLeaseClient
+{
+public:
+    BlobLeaseClient(const BlobClient& blobClient, std::string leaseId) : blob_client(&blobClient), lease_id(std::move(leaseId)) {}
+
+    BlobLeaseClient(const BlobContainerClient& blobContainerClient, std::string leaseId) : blob_container_client(&blobContainerClient), lease_id(std::move(leaseId)) {}
+
+    static std::string CreateUniqueLeaseId();
+
+    Azure::Core::Http::Response<BlobLease> Acquire(int duration, const AcquireLeaseOptions& options = AcquireLeaseOptions());
+
+    Azure::Core::Http::Response<BlobLease> Renew(const RenewLeaseOptions& options = RenewLeaseOptions());
+
+    Azure::Core::Http::Response<BlobLease> Release(const ReleaseLeaseOptions& options = ReleaseLeaseOptions());
+
+    Azure::Core::Http::Response<BlobLease> Change(const std::string& proposedId, const ChangeLeaseOptions& options = ChangeLeaseOptions());
+
+    Azure::Core::Http::Response<BlobLease> Break(const BreakLeaseOptions& options = BreakLeaseOptions());
+
+private:
+    const BlobContainerClient* blob_container_client = nullptr;
```

Comment:
>nullptr [](start = 55, length = 7)

Are you suggesting this is a weak pointer? Why not use smart pointers?

Reply:
Because we never know the `blob_client` or `blob_container_client` is a raw pointer or shared pointer or local object. We cannot make such assumption.

Reply:
The lifetime of the client is a big problem, maybe we should make a local copy?

Reply:
Maybe we indeed should. Passing pointers around is the last thing we want our customers to do.

Reply:
Maybe we can leverage `std::reference_wrapper`. In this case, if the argument is passed by value, we make a copy. Otherwise, no matter it's a `std::shared_ptr` or `std::cref`, no copy.